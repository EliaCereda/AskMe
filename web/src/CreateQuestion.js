import React, {Component} from "react";

class CreateQuestion extends Component {
    constructor(props) {
        super(props);

        this.state = {title: '', body: ''};

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        const state = {};

        state[event.target.name] = event.target.value;

        this.setState(state);
    }

    handleSubmit(event) {
        this.props.onCreate(this.state);

        this.setState({title: '', body: ''});

        event.preventDefault();
    }

    render() {
        return (
            <div className="CreateQuestion">
                <h2>Ask a new question</h2>
                <form onSubmit={this.handleSubmit}>
                    <input type="text" name="title" value={this.state.title} placeholder="Title…" onChange={this.handleChange} required/>
                    <br/>
                    <textarea name="body" value={this.state.body} placeholder="Write something…" onChange={this.handleChange} required/>
                    <br/>
                    <input type="submit" value="Submit"/>
                </form>
            </div>
        );
    }
}

export default CreateQuestion;
