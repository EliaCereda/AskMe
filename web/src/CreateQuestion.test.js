import React from 'react';
import CreateQuestion from "./CreateQuestion";

describe('CreateQuestion', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<CreateQuestion/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('triggers login event', () => {
        const title = "Questo è il titolo della domanda?";
        const body = "Si. E questo è il corpo della domanda.";

        const handleLogin = sinon.spy();
        const wrapper = shallow(<CreateQuestion onCreate={handleLogin}/>);

        wrapper.find('input[type="text"]').simulate('change', {target: {name: 'title', value: title}});
        wrapper.find('textarea').simulate('change', {target: {name: 'body', value: body}});
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {}
        });

        expect(handleLogin.withArgs({title, body}).calledOnce).toBe(true);
    });
});
