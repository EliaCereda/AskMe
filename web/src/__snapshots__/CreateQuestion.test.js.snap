// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateQuestion renders correctly 1`] = `
<div
  className="CreateQuestion"
>
  <h2>
    Ask a new question
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      name="title"
      onChange={[Function]}
      placeholder="Title…"
      required={true}
      type="text"
      value=""
    />
    <br />
    <textarea
      name="body"
      onChange={[Function]}
      placeholder="Write something…"
      required={true}
      value=""
    />
    <br />
    <input
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
